dali/pipeline/data/tensor_list.h

```diff
@@ -112,9 +93,10 @@ class DLL_PUBLIC TensorList : private Buffer<Backend> {
     this->SetLayout(other.GetLayout());
 
     use_copy_kernel &= (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned()) &&
-                       (std::is_same<Backend, GPUBackend>::value || pinned_);
-    type_.template Copy<Backend, SrcBackend>(this->raw_mutable_data(), other.raw_data(),
-                                             this->size(), stream, use_copy_kernel);
+                       (std::is_same<Backend, GPUBackend>::value || is_pinned());
+    type_info().template Copy<Backend, SrcBackend>(unsafe_raw_mutable_data(*this),
```

Comment:
Nitpick: type_info is no longer a dependent name, so template disambiguation is not necessary (unless some buggy compiler thinks otherwise, of course).
```suggestion
    type_info().Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
```